File: src/app/Employee/employee.ts
//import { Status } from "../status/status";

import { Status } from "../status/status";

export interface Employee {
    profileImage:string,
    firstName:string,
    lastName:string,
    email:string,
    location:string,
    department:string,
    jobTitle:string,
    id:string,
    joinDate:Date,
    status:Status,
    managerId:string
}

// export class Employee
// {
//     firstName:string;
//     lastName:string;
//     email:string;
//     location:string;
//     department:string;
//     jobTitle:string;
//     id:string;
//     joinDate:Date;
//     status :Status;

//     constructor(employeeData:any={}){
//         this.firstName = employeeData.firstName,
//         this.lastName = employeeData.lastName,
//         this.id = employeeData.id,
//         this.department = employeeData.department,
//         this.jobTitle = employeeData.jobTitle,
//         this.email = employeeData.email,
//         this.location = employeeData.location,
//         this.joinDate = employeeData.joinDate,
//         this.status = employeeData.status
//     }
// }

File: Models/Employee.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics.CodeAnalysis;

namespace Models
{
    public class Employee
    {

        public string Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }

        public string DateOfBirth { get; set; }
        public string Email { get; set; }
        public string MobileNumber { get; set; }
        public string JoinDate { get; set; }
        public string JobTitle { get; set; }
        public string Location { get; set; }
        public string Department { get; set; }

        public string ManagerId { get; set; }
        public string Project { get; set; }
        public Status Status { get; set; }
    }
}
